### PR TITLE
[admin] delete user from grid

### PR DESCRIPTION
**Pull request by cmangeat**
linked to https://github.com/camptocamp/c2cgeoform/pull/117
https://github.com/camptocamp/c2cgeoform/pull/117/commits/17def80d7d822cb14ae5a89e1fc033b85622b08c

### CODE REVIEW

**Comment by arnaud-morvan** [comment]
@sbrunner: Why do you closed this PR ? For me it should be rebased on master.

**Comment by sbrunner** [comment]
I delete the branch use_node_modules_jinja2, and probably GitHub automatically close this pull request ...

File: admin/requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
--e git://github.com/camptocamp/c2cgeoform.git@9718b29415707d08fecfdced62cb165a64826f2e#egg=c2cgeoform
+-e git://github.com/camptocamp/c2cgeoform.git@123561e963e43b765040e973c92599159cbd2170#egg=c2cgeoform
```
Comment (arnaud-morvan) [inline admin/requirements.txt]: need to be updated after https://github.com/camptocamp/c2cgeoform/pull/117 is merged

File: admin/c2cgeoportal_admin/views/roles.py [hunk 2]
```diff
@@ -35,3 +35,7 @@ def view(self):
                  renderer="../templates/edit.jinja2")
     def save(self):
         return super().save()
+
+    @view_config(route_name='c2cgeoform_delete')
```
Comment (arnaud-morvan) [inline admin/c2cgeoportal_admin/views/roles.py]: Should be updated accordingly to routes in c2cgeoform after merge of camptocamp/c2cgeoform#117

File: admin/acceptance_tests/user_test.py [hunk 3]
```diff
@@ -48,6 +49,21 @@ def test_view_edit(self, dbsession, test_app):
...
+        from c2cgeoportal_commons.models.main import User
+        user_id = dbsession.query(User). \
+            filter(User.username == 'babar_9'). \
+            one().id
+        from c2cgeoportal_admin.views.users import UserViews
+        req = DummyRequest(dbsession=dbsession)
+        req.referer = 'this is a test, TestUser-test_delete'
+        req.matchdict.update({'id': str(user_id)})
+
+        UserViews(req).delete()
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/user_test.py]: I would prefer we pass through the entire app using test_app

File: admin/acceptance_tests/user_test.py [hunk 4]
```diff
@@ -48,6 +49,21 @@ def test_view_edit(self, dbsession, test_app):
...
+
+        user = dbsession.query(User).filter(User.id == user_id).one_or_none()
+        assert user is None
+
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/user_test.py]: For me test of return status and location header is missing here

File: admin/acceptance_tests/user_test.py [hunk 5]
```diff
@@ -150,3 +166,39 @@ def test_selenium(self, dbsession, selenium):
...
+    @pytest.mark.usefixtures("selenium", "selenium_app")
+    def test_delete_selenium(self, dbsession, selenium):
+        from c2cgeoportal_commons.models.main import User
+        user_id = dbsession.query(User). \
+            filter(User.username == 'babar_13'). \
+            one().id
+        selenium.get('http://127.0.0.1:6543' + '/user/')
+        elem = selenium.find_element_by_xpath("//div[@class='infos']")
+        assert 'Showing 1 to 10 of 23 entries' == elem.text
+        main_window_handle = selenium.current_window_handle
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/user_test.py]: main_window_handle seems unused

File: admin/acceptance_tests/user_test.py [hunk 6]
```diff
@@ -150,3 +166,39 @@ def test_selenium(self, dbsession, selenium):
...
+    # in order to make this work, had to install selenium gecko driver
+    @pytest.mark.usefixtures("selenium", "selenium_app")
+    def test_delete_selenium(self, dbsession, selenium):
+        from c2cgeoportal_commons.models.main import User
+        user_id = dbsession.query(User). \
+            filter(User.username == 'babar_13'). \
+            one().id
+        selenium.get('http://127.0.0.1:6543' + '/user/')
+        elem = selenium.find_element_by_xpath("//div[@class='infos']")
+        assert 'Showing 1 to 10 of 23 entries' == elem.text
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/user_test.py]: is this related with delete ?

File: admin/acceptance_tests/user_test.py [hunk 7]
```diff
@@ -150,3 +166,39 @@ def test_selenium(self, dbsession, selenium):
...
+        main_window_handle = selenium.current_window_handle
+        elem = selenium.find_element_by_xpath("//button[@title='Refresh']/following-sibling::*")
+        elem.click()
+        elem = selenium.find_element_by_xpath("//a[contains(.,'50')]")
+        elem.click()
+        elem = selenium.find_element_by_xpath("//a[contains(@href,'" +
+                                              str(user_id) +
+                                              "/edit')]/following-sibling::*")
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/user_test.py]: Are you sure that delete will stay the edit following-sibling, this is hard to understand, is it not possible to get the delete button directly ?

File: admin/acceptance_tests/user_test.py [hunk 8]
```diff
@@ -150,3 +166,39 @@ def test_selenium(self, dbsession, selenium):
...
+        elem.click()
+        selenium.switch_to_alert().accept()
+
+        selenium.switch_to_default_content()
+        selenium.refresh()
+        elem = selenium.find_element_by_xpath("//div[@class='infos']")
+        assert 'Showing 1 to 10 of 22 entries' == elem.text
+        elem = selenium.find_element_by_xpath("//button[@title='Refresh']/following-sibling::*")
+        elem.click()
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/user_test.py]: Is it stricly neccessary to do such many refreshes

File: admin/acceptance_tests/user_test.py (hunk 6 above)
Comment (cmangeat) [reply admin/acceptance_tests/user_test.py]: Ensuring that there 23 entries before delete, then 22 after delete, also that all entries can be read on the same page (50 per pages, i.e. more than 23)

File: admin/acceptance_tests/user_test.py (hunk 7 above)
Comment (cmangeat) [reply admin/acceptance_tests/user_test.py]: Need to add a ref for this in template at c2cgeoform template side, no sure that we want to.

File: admin/acceptance_tests/user_test.py (hunk 8 above)
Comment (cmangeat) [reply admin/acceptance_tests/user_test.py]: Seems that gecko driver need a little time to shift focus exiting alert window. I Prefer calling refresh than adding tempo.

File: admin/acceptance_tests/user_test.py (hunk 7 above)
Comment (arnaud-morvan) [reply admin/acceptance_tests/user_test.py]: We will have tu put a class on it for the icon, this could be addressed in this PR too.

Comment (cmangeat) [reply admin/acceptance_tests/user_test.py]: Icon don't carry id ! e.g. user.id.

**Comment by arnaud-morvan** [comment]
Superseeded by #3208
